### DISCUSSION TITLE
Event handler that triggers by record create/update

### DISCUSSION THREAD

**Discussion by helander** [pocketbase/pocketbase#3196] [Q&A] [answered]
I can have a trigger that triggers if a database record is created or changed using the CRUD API hooks, but what if the change did not come from a CRUD API call?
Let’s say that in some arbitrary event hook a collection record is updated, and I want to “monitor” this by some other “unrelated” event hook. Is this possible?

**Comment by ganigeorgiev** [comment, marked as answer]
I'm not sure if that is what you are looking for, but you can listen to all db model changes (not just those from the api requests) using the [`OnModel*` Go hooks](https://pocketbase.io/docs/go-event-hooks/#db-hooks) (or the equivalent [`onModel*` JS hooks](https://pocketbase.io/docs/js-event-hooks/#db-hooks)).

The above hooks are triggered for every create/update/delete operation originated from `app.Dao()`/`$app.dao()` (eg. from `app.Dao().SaveRecord(record)`).

**Reply by helander** [reply]
Ok, thanks. Without having studied the OnModel* events my thought was that they were about changes of the data model, i.e. adding,removing,changing the set of fields in a collection and not its specific records. But I am happy that they provide what I asked for.
